api/app/models/base.py

```diff
@@ -49,13 +50,14 @@ def list(cls, db_session: "Session") -> list[Type["Base"]]:
 
     @classmethod
     def read(
-        cls, db_session: "Session", uid: Optional[UUID] = None, **kwargs
+        cls, db_session: "Session",
+        identifier: Union[str, UUID], **kwargs
     ) -> Type[Self]:
         del kwargs
-        if uid is None:
-            raise ValueError("uid is required")
-        with db_session as session:
-            return session.query(cls).where(cls.uid == uid).one()
+        identifier = cls.to_uid(identifier)
```

Comment:
keeping the same behavior here, though now we take a lot of different possible identifiers